src/cpuinfo_x86.c

```diff
@@ -125,6 +125,41 @@ static bool HasTmmOsXSave(uint32_t xcr0_eax) {
                                MASK_ZMM16_31 | MASK_XTILECFG | MASK_XTILEDATA);
 }
 
+static bool HasSecondFMA(uint32_t model) {
+    // Skylake server
+    if (model==0x55) {
+        char proc_name[49] = {0};
+        FillX86BrandString(proc_name);
+        // detect Xeon
+        if (proc_name[9]=='X') {
```

Comment:
Is testing individual character the official way? It feels like upcoming names could match and trigger false positives.

Reply:
All of this code is within the `model==0x55` conditional so it is not vulnerable to future branding changes unless they were based on the Skylake Server microarchitecture, which seems exceedingly unlikely.

In C++ implementations of this code, I do substring matching but in the C versions I just test single characters since I believe I have complete knowledge of the relevant processor names (including off-roadmap ones), in which case it should be sufficient.

Would you prefer I test multiple characters directly (test for 'X', 'e', 'o', 'n' not just 'X'), use `strncmp`, or use `strstr`?  I generally avoid the latter since it doesn't have an `n` version.